src/components/main-layout/ContentPanelTimer.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useContext, useEffect, useRef, useState } from "react";
+
+import styles from "./layout.module.css";
+import { SimulariumContext } from "../../simulation/context";
+import { PageContent, Module } from "../../types";
+
+import classNames from "classnames";
+import { isEqual } from "lodash";
+import ContentPanel from "./ContentPanel";
+
+export interface ContentPanelProps {
+    pageContent: PageContent;
+    currentModule: Module;
+}
+
+// these three states are used to avoid flashing the content before the
+// opacity transition starts
+enum RenderState {
+    NewContent = 1,
+    StartFade = 2,
+    FullyVisible = 3,
+}
+
+const ContentPanelTimer: React.FC<ContentPanelProps> = ({
+    pageContent,
+    currentModule,
+}) => {
+    const [renderState, setRenderState] = useState(RenderState.NewContent);
+    const previousContentRef = useRef(pageContent);
+
+    useEffect(() => {
+        // There are pages that have the same written content, so flashing the
+        // panel is more distracting than helpful. However, since this Ref is defaulted
+        // to the first page content, this check will always be true on the first page, and
+        // will keep the content from being rendered, so it needs to be excluded
+        if (!contentJustChanged && page !== 1) {
```

Comment:
Feels strange to have a `useEffect` that includes `contentJustChanged` before `contentJustChanged` is declared in the component